cg/meta/store/mip_rna.py

```diff
@@ -1,102 +1,176 @@
-            'archive': False,
-            })
+    data = [
+        {
+            "path": file["path"],
+            "tags": sorted(
+                list(
+                    set(
+                        [
+                            file["format"],
+                            file["id"],
+                            file["step"],
+                            file["tag"],
+                            "rd-rna",
+                        ]
+                    )
+                    - set([None])
+                )
+            ),
+            "archive": False,
+        }
+        for file in deliverables["files"]
+    ]
 
     return data
+
+
+def parse_config(data: dict) -> dict:
+    """Parse MIP config file.
+
+    Args:
+        data (dict): raw YAML input from MIP analysis config file
+
+    Returns:
+        dict: parsed data
+    """
+    return {
+        "email": data.get("email"),
+        "case": data["case_id"],
+        "samples": [
+            {"id": sample_id, "type": analysis_type}
+            for sample_id, analysis_type in data["analysis_type"].items()
+        ],
+        "is_dryrun": bool("dry_run_all" in data),
+        "out_dir": data["outdata_dir"],
+        "priority": data["slurm_quality_of_service"],
+        "sampleinfo_path": data["sample_info_file"],
+    }
+
+
+def parse_sampleinfo(data: dict) -> dict:
+    """Parse MIP sample info file (RNA).
+
+    Args:
+        data (dict): raw YAML input from MIP qc sample info file (RNA)
+
+    Returns:
+        dict: parsed data
+    """
+    case = data["case"]
+
+    sampleinfo_data = {
+        "date": data["analysis_date"],
+        "is_finished": data["analysisrunstatus"] == "finished",
+        "case": case,
+        "version": data["mip_version"],
+    }
+
+    return sampleinfo_data
+
+
+def get_case(bundle_obj, status):
+    """ Get a case from the status database """
+    case_obj = status.family(bundle_obj.name)
+    return case_obj
+
+
+def reset_case_action(case_obj):
+    """ Resets action on case """
+    case_obj.action = None
+
+
+def add_new_complete_analysis_record(bundle_data, case_obj, status, version_obj):
+    """Function to create and return a new analysis database record"""
+    pipeline = case_obj.links[0].sample.data_analysis
+    pipeline = pipeline if pipeline else "mip"
+
+    if status.analysis(family=case_obj, started_at=version_obj.created_at):
+        raise AnalysisDuplicationError(
+            f"Analysis object already exists for {case_obj.internal_id} {version_obj.created_at}"
+        )
+
+    new_analysis = status.add_analysis(
+        pipeline=pipeline,
+        version=bundle_data["pipeline_version"],
+        started_at=version_obj.created_at,
+        completed_at=dt.datetime.now(),
+        primary=(len(case_obj.analyses) == 0),
+    )
+    new_analysis.family = case_obj
+    return new_analysis
+
+
+def include_files_in_housekeeper(bundle_obj, hk_api, version_obj):
```

Comment:
This def seem to be agnostic to pipeline. Could it be placed in a store/base.py or similiar to be imported and used by multiple pipelines?

Reply:
https://github.com/Clinical-Genomics/cg/pull/570/commits/3af147be0325c5ea81a6e1e96038e421ab86c3fc